src/audio/volume/volume.c

```diff
@@ -712,6 +834,59 @@ static int volume_cmd(struct comp_dev *dev, int cmd, void *data,
 	}
 }
 
+/* CONFIG_IPC_MAJOR_3 */
+#elif CONFIG_IPC_MAJOR_4
+/**
+ * \brief Used to pass standard and bespoke commands (with data) to component.
+ * \param[in,out] dev Volume base component device.
+ * \param[in] cmd Command type.
+ * \param[in,out] data Control command data.
+ * \return Error code.
+ */
+static int volume_cmd(struct comp_dev *dev, int cmd, void *data,
+		      int max_data_size)
+{
+	struct ipc4_peak_volume_config *cdata = ASSUME_ALIGNED(data, 8);
+	struct vol_data *cd = comp_get_drvdata(dev);
+	uint32_t channels_count = cd->base.audio_fmt.channels_count;
+	int i;
+
+	comp_dbg(dev, "volume_cmd()");
+
+	dcache_invalidate_region(cdata, sizeof(*cdata));
```

Comment:
is this needed because the config can be used on different cores?

Reply:
oh, I think this has been updated at the IPC layer (so that IPC layer does all the cache mgmt) instead of here so the cache inv is not needed here.

Reply:
yeap, it was updated for creating comp but this case is still affected.

Anyway removed dcache and ipc4 handler needs to be updated in another PR.
For now it doesnt matter because looks like volume_cmd() is never used for IPC4 because commit 6a297b4ed7b8 added set large config but also removed cmd translation:
module set_large_config <-------> component cmd